### DISCUSSION TITLE
Analytics on Assets

### DISCUSSION THREAD

**Discussion by st11x** [directus/directus#12801] [Archive]
Hi,

I would like to keep track of how many times a PDF file has been downloaded for example. How can I achieve that?

Do hooks do that? They have read events, but I am thinking it's more about the asset getting read in Directus on top of the file getting downloaded.

Do I have to write a custom end point? I want to avoid having to write the code to stream the file content in the end point.

thanks
  Matt

**Comment by rijkvanzanten** [comment]
There's currently no "asset downloaded" hook, so I think this will have to be a custom endpoint 🤔

**Reply by st11x** [reply]
Got it. Thanks.
